src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -3734,6 +3810,124 @@ private AssociativeNode DFSEmitSplitAssign_AST(AssociativeNode node, ref List<As
+        ///     -> t = a.%set_x(10)
+        ///     
+        /// a.x.y = b + c 
+        ///     -> a.x.%set_y(b + c)
+        ///     
+        /// a.x[0] = 10
+        ///     ->  tval = a.%get_x()
+        ///         tval[0] = 10         
+        ///         tmp = a.%set_x(tval)
+        /// </summary>
+        /// <param name="astList"></param>
+        /// <returns></returns>
+        private List<AssociativeNode> TransformLHSIdentList(List<AssociativeNode> astList)
+        {
+            if (null != astList && astList.Count > 0)
+            {
+                List<AssociativeNode> newAstList = new List<AssociativeNode>();
+                foreach (AssociativeNode node in astList)
+                {
+                    BinaryExpressionNode bNode = node as BinaryExpressionNode;
+                    if (bNode == null)
+                    {
+                        newAstList.Add(node);
+                    }
+                    else
+                    {
+                        bool isLHSIdentList = bNode.LeftNode is IdentifierListNode;
+                        if (!isLHSIdentList)
+                        {
+                            newAstList.Add(node);
+                        }
+                        else
+                        {
+                            IdentifierNode lhsTemp = new IdentifierNode(Constants.kTempVar);
+
+                            IdentifierListNode identList = bNode.LeftNode as IdentifierListNode;
+                            Validity.Assert(identList != null);
+
+                            AssociativeNode argument = bNode.RightNode;
+
+                            IdentifierNode identFunctionCall = identList.RightNode as IdentifierNode;
+                            string setterName = ProtoCore.DSASM.Constants.kSetterPrefix + identList.RightNode.Name;
+                            bool isArrayIndexed = identFunctionCall.ArrayDimensions != null;
+                            if (isArrayIndexed)
+                            {
+                                // a.x[0] = 10
+                                //      tval = a.%get_x()
+                                string getterName = ProtoCore.DSASM.Constants.kGetterPrefix + identList.RightNode.Name;
+                                ProtoCore.AST.AssociativeAST.FunctionCallNode fcall = new ProtoCore.AST.AssociativeAST.FunctionCallNode();
+                                fcall.Function = new IdentifierNode(identList.RightNode.Name);
+                                fcall.Function.Name = getterName;
+
+                                IdentifierListNode identList1 = new IdentifierListNode();
+                                identList1.LeftNode = identList.LeftNode;
+                                identList1.RightNode = fcall;
+                                BinaryExpressionNode bnodeGet = new BinaryExpressionNode(
+                                    lhsTemp,
+                                    identList1
+                                    );
```

Comment:
By default the operator of this binary expression is `Operator.none`, is that expected?